Public/Src/Cache/ContentStore/Distributed/Sessions/RecordingStream.cs

```diff
@@ -14,21 +14,36 @@ internal class RecordingStream : Stream
     {
         private readonly byte[] _recordedBytes;
         private readonly Stream _inner;
-        private readonly long _capacity;
+        private readonly long? _capacity;

```

Comment:
Minor and arguable, but in this case, where nullity is simple and fixed, I would just use _capacity = -1 as a signal, and avoid the Nullable layer.